jina/peapods/zmq/__init__.py

```diff
@@ -305,10 +365,24 @@ async def send_message(self, msg: 'Message', sleep: float = 0, **kwargs):
         :param kwargs: keyword arguments
         """
         # await asyncio.sleep(sleep)  # preventing over-speed sending
-        try:
-            num_bytes = await send_message_async(
-                self.out_sock, msg, **self.send_recv_kwargs
+        if self.args.dynamic_out_routing:
+            await self._send_message_dynamic(msg)
+        else:
+            self._send_message_via(self.out_sock, msg)
+
+    async def _send_message_dynamic(self, msg: 'Message'):
+        tasks = []
+        for routing_graph, out_sock in self.get_dynamic_next_routes(msg):
+            new_message_proto = msg.proto
+            new_message_proto.envelope.routing_graph.CopyFrom(routing_graph.proto)
+            tasks.append(
+                self._send_message_via(out_sock, Message.from_proto(new_message_proto))
             )
+        await asyncio.gather(*tasks)
```

Comment:
Can you please double-check, that this is the Jina conform way of doing it? @deepankarm @JoanFM Thanks!

Reply:
I think so, but @deepankarm  is the asyncio master